lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -219,7 +239,11 @@ private function getMariaDb1027ColumnDefault(MariaDb1027Platform $platform, ?str
         }
 
         if (preg_match('/^\'(.*)\'$/', $columnDefault, $matches)) {
-            return stripslashes(str_replace("''", "'", $matches[1]));
+            return str_replace(
```

Comment:
Maybe you can use `strtr` with array?

Reply:
Actually, `strtr` will give a more correct result:
```php
$str = <<<'STR'
\'\'
STR
;

$tr = [
    "\\'" => "'",
    "''"  => "'",
];

$replaced1 = str_replace(array_keys($tr), array_values($tr), $str);
$replaced2 = strtr($str, $tr);

var_dump($replaced1);
// string(1) "'"

var_dump($replaced2);
// string(2) "''"
```
Going to rework and add to the test.

Reply:
The scenario above is actually practically impossible since MariaDB duplicates single quotes instead of prepending a backslash. I added a similar test which somehow passes with both implementations and switched to the proposed implementation anyways.